### PR TITLE
RHPAM-1700 Column sorting is not retained after filters are applied i…

### PR DESCRIPTION
**Pull request by bxf12315**
…n task inbox business central

### CODE REVIEW

**Comment by bxf12315** [comment]
@cristianonicolai Please review it,

**Comment by cristianonicolai** [comment]
@bxf12315 can you please make sure you explicitly set that there is a dependency between PRs, otherwise things can get merge and break the build. Also, if you could add some tests in here too.

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-service/uberfire-widgets-service-api/src/main/java/org/uberfire/ext/services/shared/preferences/GridSortedColumnPreference.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        if (ascending != that.ascending) {
+            return false;
+        }
+        return dataStoreName != null ? dataStoreName.equals(that.dataStoreName) : that.dataStoreName == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = dataStoreName != null ? dataStoreName.hashCode() : 0;
+        result = 31 * result + (ascending ? 1 : 0);
```
Comment (cristianonicolai) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-service/uberfire-widgets-service-api/src/main/java/org/uberfire/ext/services/shared/preferences/GridSortedColumnPreference.java]:
```suggestion
        result = 31 * result + (ascending ? 1 : 0);
        result = ~~result;
```

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-service/uberfire-widgets-service-api/src/main/java/org/uberfire/ext/services/shared/preferences/GridSortedColumnPreference.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        if (ascending != that.ascending) {
+            return false;
+        }
+        return dataStoreName != null ? dataStoreName.equals(that.dataStoreName) : that.dataStoreName == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = dataStoreName != null ? dataStoreName.hashCode() : 0;
```
Comment (cristianonicolai) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-service/uberfire-widgets-service-api/src/main/java/org/uberfire/ext/services/shared/preferences/GridSortedColumnPreference.java]:
```suggestion
        int result = dataStoreName != null ? dataStoreName.hashCode() : 0;
        result = ~~result;
```

**Comment by bxf12315** [comment]
@cristianonicolai Done

**Comment by cristianonicolai** [comment]
@ederign please merge this and https://github.com/kiegroup/jbpm-wb/pull/1277 together thanks.

**Comment by cristianonicolai** [comment]
@ederign ping :)

**Pull request merged**